app/src/main/java/org/mozilla/tv/firefox/webrender/WebRenderFragment.kt

```diff
@@ -211,4 +212,11 @@ class WebRenderFragment : EngineViewLifecycleFragment(), Session.Observer {
         }
         return false
     }
+
+    /**
+     * Called before the [EngineView] is shown
+     */
+    fun onShowEngineView() {
+        engineView?.focusedDOMElement?.cache()
```

Comment:
We need to explain why we cache here and specifically here. Basically, the gist of your commit summary (cache the element at the last possible second) should really be in the code base as comments instead.